bot/exts/code_jams/_flows.py

```diff
@@ -0,0 +1,209 @@
+    Finally, a view of Team Announcement will be sent.
+    """
+    team_leaders = await ctx.guild.create_role(name=TEAM_LEADER_ROLE_NAME, colour=TEAM_LEADERS_COLOUR)
+    await _creation_utils.create_team_leader_channel(ctx.guild, team_leaders)
+    jam_api_format = {"name": f"Summer Code Jam {datetime.now().year}", "ongoing": True, "teams": []}
+    for team_name, team_members in teams.items():
+        team_role = await _creation_utils.create_team_role(
+            ctx.guild,
+            team_name,
+            team_members,
+            team_leaders
+        )
+        team_channel_id = await _creation_utils.create_team_channel(ctx.guild, team_name, team_role)
+        jam_api_format["teams"].append(
+            {
+                "name": team_name,
+                "users": [
+                    {"user_id": entry["member"].id, "is_leader": entry["is_leader"]} for entry in team_members
+                ],
+                "discord_role_id": team_role.id,
+                "discord_channel_id": team_channel_id
+            }
+        )
+    await bot.code_jam_mgmt_api.post("codejams", json=jam_api_format)
+    success_embed = Embed(
+        title=f"Successfully created Code Jam with {len(teams)} teams",
+        colour=discord.Colour.green(),
+        description="Would you like send out the team announcement?"
+    )
+    success_embed.set_footer(text="Code Jam team generation")
+    team_info_view = JamTeamInfoConfirmation(bot, ctx.guild, ctx.author)
+    await ctx.send(
+        embed=success_embed,
+        view=team_info_view
+    )
+
+
+async def deletion_flow(
+        category_channels: dict[discord.CategoryChannel: list[discord.TextChannel]],
+        roles: list[discord.Role]
+) -> None:
+    """
+    The Code Jam Team and Role deletion flow.
+
+    The "flow" will delete the channels in each category, and the category itself.
+    Then it'll delete all the Team roles, leaving the CJ Leaders related channel, and Roles intact.
+    """
+    for category, channels in category_channels.items():
+        for channel in channels:
+            await channel.delete(reason="Code jam ended.")
+        await category.delete(reason="Code jam ended.")
+    for role in roles:
+        await role.delete(reason="Code Jam ended.")
+
+
+async def move_flow(
+        bot: SirRobin,
+        new_team_name: str,
+        ctx: commands.Context,
+        member: discord.Member
+) -> None:
+    """Move participant from one team to another by issuing an HTTP request to the Code Jam Management system."""
+    # Query the team the user has to be moved to
+    try:
+        team_to_move_in = await bot.code_jam_mgmt_api.get(
+            "teams/find",
+            params={"name": new_team_name},
+            raise_for_status=True
+        )
+    except ResponseCodeError as err:
+        if err.response.status == 404:
+            await ctx.send(f":x: Team `{new_team_name}` does not exists in the database!")
```

Comment:
Let's change it "does not exist in current jam". It's not actually correct currently, because it may exist in DB under another jam.